test/TestProjects/MgmtScopeResource/Generated/Extensions/ArmResourceExtensions.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using Azure.ResourceManager.Core;
+
+namespace MgmtScopeResource
+{
+    /// <summary> A class to add extension methods to ArmResource. </summary>
+    public static partial class ArmResourceExtensions
+    {
+        #region PolicyAssignment
+        /// <summary> Gets an object representing a PolicyAssignmentCollection along with the instance operations that can be performed on it. </summary>
+        /// <param name="armResource"> The <see cref="ArmResource" /> instance the method will execute against. </param>
+        /// <returns> Returns a <see cref="PolicyAssignmentCollection" /> object. </returns>
+        public static PolicyAssignmentCollection GetPolicyAssignments(this ArmResource armResource)
```

Comment:
Do we control somehow that method names in different extensions don't clash with each other?

Reply:
If you mean the method name between the `ArmResourceExtensions` and other extensions, the generator would remove the method from other extensions if it is already in ArmResourceExtensions. For other extensions, we prefer to give them the same name, so user experience is consistent. For instance, we have `subscription.GetAll()` and `managementGroup.GetAll()` instead of the original `subscription.GetAllBySubscription()` and `managementGroup.GetAllByManagementGroup()`.
 
If you mean method names for the same extension across different RPs, I think it equals to the question that do we control resource name conflicts across different RPs as the method names are just derived from resource names. @ArcturusZhang did some work to avoid resource name conflicts in the same RP but I don't think we have the checks for all RPs. One practice in API review is to avoid single word as class names which are likely to cause name conflicts across different RPs.